src/core/config/directives/discriminate.rs

```diff
@@ -0,0 +1,29 @@
+use serde::{Deserialize, Serialize};
+use tailcall_macros::{DirectiveDefinition, InputDefinition, MergeRight};
+
+use crate::core::is_default;
+
+#[derive(
+    Serialize,
+    Deserialize,
+    Clone,
+    Debug,
+    PartialEq,
+    Eq,
+    schemars::JsonSchema,
+    DirectiveDefinition,
+    InputDefinition,
+    MergeRight,
+)]
+#[directive_definition(locations = "FieldDefinition")]
+#[serde(deny_unknown_fields)]
+///
+/// The `@discriminate` directive is used to drive Tailcall discriminator to use
+/// a field of an object to resolve the type. For example with the directive
+/// applied on a field `@discriminate(field: "type")` and the given value
+/// `{"foo": "bar", "type": "Buzz"}` the resolved type of the object will be
+/// `Buzz`.
+pub struct Discriminate {
```

Comment:
you need to add the new definition to https://github.com/tailcallhq/tailcall/blob/ccba6068a4fa45317ac7e2d220c363e466b4c88e/src/core/config/config.rs#L638 otherwise it's not populated into tailcall's schema

Reply:
done